### PR TITLE
Ceph plugin

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: keyv/Ceph.cpp [hunk 1]
```diff
@@ -32,262 +35,215 @@ lunchbox::PluginRegisterer<Ceph> registerer;
...
-                std::cerr << "Finish write failed: " << ::strerror(-write)
-                          << std::endl;
-                ok = false;
-            }
-            delete _writes.front();
-            _writes.pop_front();
-        }
-        return ok;
+inline void Ceph::getValues(const lunchbox::Strings& keys,
+                            const ConstValueFunc& func) const
```
Comment (hernando) [inline keyv/Ceph.cpp]: Is it possible to refactor this function and takeValues to extract the common code?

File: tests/Map.cpp [hunk 2]
```diff
@@ -285,6 +291,96 @@ void benchmark(const std::string& uriStr, const uint64_t queueDepth,
...
+                            const size_t valueSize)
+{
+    static std::string lastURI;
+    if (uriStr != lastURI)
+    {
+        std::cout
+            << " #thr ,     size,  writes/s,     MB/s,  reads/s,      MB/s"
+            << std::endl;
+        lastURI = uriStr;
+    }
```
Comment (hernando) [inline tests/Map.cpp]: Move the printing to the caller to avoid the static variable?

**Comment by tribal-tec** [comment]
Couldn't you have it reviewed in the last 10 months? I just rebased it to finally merge it. I didn't type any code here, so you really want me to fix your concerns?

**Comment by hernando** [comment]
Yes, fix everything!

Most of it is just comments, but at least the documentation and the changelog should be updated.

**Comment by tribal-tec** [comment]
Regarding performance: I have no idea honestly. Don't know why it was async before, now it's not. If it makes a difference or not, no idea. The two involved people are gone now, so as long as you or somebody else knows those things, it's a black box to me. I'm just here for merging :)

**Comment by favreau** [comment]
Before Mohamed left, I spent some time with him setting up the solution on our system. Ceph performance was poor and that seems to be a common issue with other installation of the product. The purpose of this PR is to test the solution on another system, but my understanding is that performance will still be an issue there. Let's see...

**Comment by tribal-tec** [comment]
ping

**Comment by favreau** [comment]
Regarding the comments, I agree with @tribal-tec. The goal of the PR is to make Mohamed's work available for deployment. We need to build the according modules asap to finally complete this task.

**Comment by hernando** [comment]
Do you mind updating the documentation of Map::Map to reflect how the URL is processed?

**Comment by tribal-tec** [comment]
Done

**Pull request merged**